### How to split by character -Character Text splitter

This splits based on a given character sequences , which defaults to "\n\n". Chunk length is measured by number of
characters.

the text is split by single character separator.
the chunk size is measured by number of characters.


In [1]:
from langchain_community.document_loaders import TextLoader  #text loader
loader = TextLoader('./speech.md')
text_document = loader.load() #load the text as document object
text_document

[Document(metadata={'source': './speech.md'}, page_content="Ladies and gentlemen,\n\nToday, we stand at the crossroads of innovation and tradition. Technology has transformed the way we live, work, and communicate. It is our responsibility to harness this power for the greater good.\n\nTogether, let's build a future that is inclusive, sustainable, and driven by knowledge.\n\nThank you.\n")]

In [6]:
from langchain_text_splitters import CharacterTextSplitter
C_splitters = CharacterTextSplitter(separator='\n\n',chunk_size=100,chunk_overlap=20)
C_splitters.split_documents(text_document)

Created a chunk of size 198, which is longer than the specified 100


[Document(metadata={'source': './speech.md'}, page_content='Ladies and gentlemen,'),
 Document(metadata={'source': './speech.md'}, page_content='Today, we stand at the crossroads of innovation and tradition. Technology has transformed the way we live, work, and communicate. It is our responsibility to harness this power for the greater good.'),
 Document(metadata={'source': './speech.md'}, page_content="Together, let's build a future that is inclusive, sustainable, and driven by knowledge.\n\nThank you.")]

it created the chunk size of 198 instead of 100 because it was not able to find the sperator(\n\n) so it build the chunk size of 198 next smallest where it found the splitter